sklearn/tests/test_multioutput.py

```diff
@@ -339,3 +348,142 @@ def test_multi_output_exceptions():
+
+    Y_multi = np.array([[int(yyy) for yyy in format(yy, '#06b')[2:]]
+                        for yy in y])
+    return X, Y_multi
+
+
+def test_classifier_chain_fit_and_predict_with_logistic_regression():
+    # Fit classifier chain and verify predict performance
+    X, Y = generate_multilabel_dataset_with_correlations()
+    classifier_chain = ClassifierChain(LogisticRegression())
+    classifier_chain.fit(X, Y)
+
+    Y_pred = classifier_chain.predict(X)
+    assert_equal(Y_pred.shape, Y.shape)
+
+    Y_prob = classifier_chain.predict_proba(X)
+    Y_binary = (Y_prob >= .5)
+    assert_array_equal(Y_binary, Y_pred)
+
+    assert_equal([c.coef_.size for c in classifier_chain.estimators_],
+                 list(range(X.shape[1], X.shape[1] + Y.shape[1])))
+
+
+def test_classifier_chain_fit_and_predict_with_linear_svc():
+    # Fit classifier chain and verify predict performance using LinearSVC
+    X, Y = generate_multilabel_dataset_with_correlations()
+    classifier_chain = ClassifierChain(LinearSVC())
+    classifier_chain.fit(X, Y)
+
+    Y_pred = classifier_chain.predict(X)
+    assert_equal(Y_pred.shape, Y.shape)
+
+    Y_decision = classifier_chain.decision_function(X)
+
+    Y_binary = (Y_decision >= 0)
+    assert_array_equal(Y_binary, Y_pred)
+
+
+def test_classifier_chain_fit_and_predict_with_sparse_data():
+    # Fit classifier chain with sparse data
+    X, Y = generate_multilabel_dataset_with_correlations()
+    X_sparse = sp.csr_matrix(X)
+
+    classifier_chain = ClassifierChain(LogisticRegression())
+    classifier_chain.fit(X_sparse, Y)
+    Y_pred_sparse = classifier_chain.predict(X_sparse)
+
+    classifier_chain = ClassifierChain(LogisticRegression())
+    classifier_chain.fit(X, Y)
+    Y_pred_dense = classifier_chain.predict(X)
+
+    assert_array_equal(Y_pred_sparse, Y_pred_dense)
+
+
+def test_classifier_chain_fit_and_predict_with_sparse_data_and_cv():
+    # Fit classifier chain with sparse data cross_val_predict
+    X, Y = generate_multilabel_dataset_with_correlations()
+    X_sparse = sp.csr_matrix(X)
+    classifier_chain = ClassifierChain(LogisticRegression(), cv=3)
+    classifier_chain.fit(X_sparse, Y)
+    Y_pred = classifier_chain.predict(X_sparse)
+    assert_equal(Y_pred.shape, Y.shape)
+
+
+def test_classifier_chain_random_order():
+    # Fit classifier chain with random order
+    X, Y = generate_multilabel_dataset_with_correlations()
+    classifier_chain_random = ClassifierChain(LogisticRegression(),
+                                       order='random',
+                                       random_state=42)
+    classifier_chain_random.fit(X, Y)
+    Y_pred_random = classifier_chain_random.predict(X)
+
+    assert_not_equal(list(classifier_chain_random.order), list(range(4)))
+    assert_equal(len(classifier_chain_random.order_), 4)
+    assert_equal(len(set(classifier_chain_random.order_)), 4)
+
+    classifier_chain_fixed = ClassifierChain(LogisticRegression(),
```

Comment:
a comment like "Random order should behave identically to the same order being specified" before this might be helpful to the reader